src/app/core/services/action-trigger-state.service.ts

```diff
@@ -29,49 +31,61 @@ export class ActionTriggerStateService {
 
   private subscribeToActionTriggerEvents(): void {
     this.subscriptions.push(
-      this.actionTriggerEventObserver.subscribeToActionTriggerCreated((actionTriggerCreatedEvent: ActionTriggerCreatedEvent) => this.addCreatedActionTrigger(actionTriggerCreatedEvent))
+      this.actionTriggerEventObserver.subscribeToActionTriggerCreated((actionTriggerCreatedEvent: ActionTriggerCreatedEvent) => this.addCreatedActionTriggerState(actionTriggerCreatedEvent))
     )
     this.subscriptions.push(
-      this.actionTriggerEventObserver.subscribeToActionTriggerUpdated((actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent) => this.updateActionTrigger(actionTriggerUpdatedEvent))
+      this.actionTriggerEventObserver.subscribeToActionTriggerUpdated((actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent) => this.updateActionTriggerState(actionTriggerUpdatedEvent))
     )
     this.subscriptions.push(
-      this.actionTriggerEventObserver.subscribeToActionTriggerDeleted((actionTriggerDeletedEvent: ActionTriggerDeletedEvent) => this.removeActionTrigger(actionTriggerDeletedEvent))
+      this.actionTriggerEventObserver.subscribeToActionTriggerDeleted((actionTriggerDeletedEvent: ActionTriggerDeletedEvent) => this.removeActionTriggerState(actionTriggerDeletedEvent))
     )
   }
 
-  private addCreatedActionTrigger(actionTriggerCreatedEvent: ActionTriggerCreatedEvent): void {
-    const actionTriggers: ActionTrigger[] = this.actionTriggerSubject.value
-    actionTriggers.push(actionTriggerCreatedEvent.actionTrigger)
-    this.actionTriggerSubject.next(actionTriggers)
-  }
-
-  private updateActionTrigger(actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent): void {
-    const actionTriggers: ActionTrigger[] = this.actionTriggerSubject.value
+  private updateActionTriggerState(actionTriggerUpdatedEvent: ActionTriggerUpdatedEvent): void {
+    const actionTriggers: ActionTrigger<KeyboardTriggerData>[] = this.actionTriggerSubject.value
     const index: number = actionTriggers.findIndex(actionTrigger => actionTrigger.id === actionTriggerUpdatedEvent.actionTrigger.id)
     if (index === -1) {
-      throw new Error(`Updated ActionTrigger does not exist in ActionTrigger state`)
+      this.openDangerSnackBar('Fail to update')
```

Comment:
I would change the message to e.g `'Failed to update shortcut'`.

Reply:
Done